consensus/config/src/committee.rs

```diff
@@ -36,13 +36,18 @@ pub struct Committee {
 }
 
 impl Committee {
-    pub fn new(epoch: Epoch, authorities: Vec<Authority>) -> Self {
+    pub fn new(epoch: Epoch, mut authorities: Vec<Authority>) -> Self {
         assert!(!authorities.is_empty(), "Committee cannot be empty!");
         assert!(
             authorities.len() < u32::MAX as usize,
             "Too many authorities ({})!",
             authorities.len()
         );
+
+        // sort the authorities by their protocol (public) key in ascending order - critical to be
+        // aligned with the SUI committee.
+        authorities.sort_by(|a1, a2| a1.protocol_key.cmp(&a2.protocol_key));
```

Comment:
The order should be provided by the caller or SUI in this case right? That would make sure we are aligned if something changes in the future on the Sui side?

Reply:
Good suggestion.

Reply:
Agreed, I was very in between this approach and the one that I went with , but it seems it makes more sense. I'll refactor. Thanks @arun-koshy